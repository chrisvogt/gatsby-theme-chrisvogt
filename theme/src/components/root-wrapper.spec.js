import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import RootWrapper from './root-wrapper'

// Mock AudioPlayer component
jest.mock('./audio-player', () => jest.fn(() => <div>MockAudioPlayer</div>))

const mockStore = configureStore([])

describe('RootWrapper', () => {
  let store

  beforeEach(() => {
    store = mockStore({
      audioPlayer: {
        soundcloudId: '123',
        isVisible: true
      }
    })
    store.dispatch = jest.fn()
  })

  it('renders children and AudioPlayer', () => {
    const component = renderer.create(
      <Provider store={store}>
        <RootWrapper>
          <div>Test Child</div>
        </RootWrapper>
      </Provider>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('passes correct props to AudioPlayer', () => {
    const AudioPlayer = require('./audio-player')
    renderer.create(
      <Provider store={store}>
        <RootWrapper>
          <div>Test Child</div>
        </RootWrapper>
      </Provider>
    )

    expect(AudioPlayer).toHaveBeenCalledWith(
      {
        soundcloudId: '123',
        isVisible: true
      },
      expect.anything()
    )
  })
})
