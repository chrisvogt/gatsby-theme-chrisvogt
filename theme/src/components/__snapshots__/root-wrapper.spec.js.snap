// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootWrapper renders children and AudioPlayer 1`] = `
[
  <div>
    Test Child
  </div>,
  <div>
    MockAudioPlayer
  </div>,
]
`;
